mungegithub/mungers/flakeyjob-reporter.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"k8s.io/test-infra/mungegithub/mungers/mungerutil"
+)
+
+// FlakeyJob is a struct that represents a single job and the flake data associated with it.
+// FlakeyJob implements the Issue interface so that it can be synced with github issues via the IssueCreator.
+type FlakeyJob struct {
+	// Name is the job's name.
+	Name string
+	// Consistency is the percentage of builds that passed.
+	Consistency *float32 `json:"consistency"`
```

Comment:
I used pointers for fields that are unmarshalled from JSON so that there is an out-of-band indicator if a field is missing from the JSON. If Consistency was not a pointer it would be set to 0 if the key is missing from the JSON which is indistinguishable from an actual consistency value of 0.